cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,13 +186,32 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      final List<String> additionalClassPaths = new ArrayList<>();
+      // Add extra jars set in cConf to additionalClassPaths and localizeResources
+      for (URI jarURI : CConfigurationUtil.getExtraJars(cConf)) {
+        LocalizeResource localizeResource;
+        String protocol = jarURI.getScheme();
+        if (protocol == null) {
+          localizeResource = new LocalizeResource(new File(jarURI.getPath()));
+        } else if ("file".equals(protocol)) {
+          localizeResource = new LocalizeResource(new File(jarURI));
+        } else {
+          localizeResource = new LocalizeResource(jarURI, false);
+        }
+        String jarName = new File(jarURI.getPath()).getName();
```

Comment:
It is not guaranteed to be able to get the filename from URI. E.g. if cdap-master is running on a Windows box (the cluster can still be linux). It may not be the case right now, but it's better to write robust code. It should be something like this: https://github.com/caskdata/cdap/blob/develop/cdap-master/src/main/java/co/cask/cdap/data/runtime/main/ExploreServiceTwillRunnable.java#L187

Maybe it's worth to put it under `Networks` util class in cdap-common.